discovery-provider/src/utils/helpers.py

```diff
@@ -14,6 +14,44 @@
 from src import exceptions
 from . import multihash
 
+def redis_restore(redis, key):
+    logger = logging.getLogger(__name__)
+    try:
+        filename = f"{key}_dump"
+        with open(filename, 'rb') as f:
+            dumped = f.read()
+            redis.restore(key, 0, dumped)
+            logger.info(f"successfully restored redis value for key: {key}")
+            return redis.get(key)
+    except FileNotFoundError as not_found:
+        logger.error(f"could not read redis dump file: {filename}")
+        logger.error(not_found)
+        return None
+    except Exception as e:
+        logger.error(f"could not perform redis restore for key: {key}")
+        logger.error(e)
+        return None
+
+def redis_dump(redis, key):
```

Comment:
this seems very memory intensive to read the entire contents of redis into memory and write to a file. have you investigated bgsave and having redis do the serialization to/from disk. this should just write to an .rdb file on the redis pod?

Reply:
it's for a specific key here, so i dont think it'll be all that bad?

Reply:
python doesn't have the same mem constraints as node so not certain if this end up causing problems, but just wanted to see if there was a shortcut here. cause there's no use for the data streamed from redis at all right? it's just written to disk?

Reply:
I think this will be on the single-MB size in practice though regardless, right @sddioulde ?

Reply:
not sure i understand the 2nd question, but this is just to prevent the accidental scenario where a redis gets flushed on a discovery node crash or restart

Reply:
i was wondering if we could use redis to do the persistence for us so we wouldn't need to use the python resources to do it. but, i can't find an easy way to dump just a specific key to disk in pyredis and bgsave dumps the entire db. so this may be a no-op

Reply:
resolving